akiraux/Akira
# WIP: Artboard as Containers

## Summary / How this PR fixes the problem?
This is the initial work to bring _basic_ container functionality to the Artboard item. 
Items created inside an artboard and items moved inside one should become children of the artboard. 
At the same time, the Layers Panel should reflect the children of an artboard and enable the user to actionate them (`hide` and `lock`). 

As always, I'd like to get some feedback on how "far" should I get with this PR. 
I'd like to progress up until the "Item should snap to Artboard's edges" task. 

I have some questions, too:

- How can the user remove an item from an Artboard? 
- If the user enlarges an existing Artboard to contain some new items, should those new items be considered as Artboard's children? In other words, can Artboards _fagocitate_  other items while growing?
- More to come...

## Steps to Test
- Create and artboard
- Create a shape **inside** the artboard
- Move the artboard from the handle. The items should move together with the artboard

## Screenshots 
![artboard_content_translation](https://user-images.githubusercontent.com/11556031/76171955-23fa9b80-6191-11ea-8cd7-1e0956324702.gif)

## Implemented Features
- [x] Creating an item inside an Artboard should make the item a child of the Artboard
- [x] Moving an artboard should move its children in a rigid way
- [x] Child coordinates are relative to Artboard origin
- [x] Deleting an item inside an Artboard should remove it from that Artboard's children
- [x] Item have hover effect when hovering over the Artboard
- [x] Item can be selected & moved by clicking on them

Known Issues / Things To Do
- [ ] Items inside Artboard could be rotated with the nob
- [ ] Artboard should not be allowed to be rotated
- [ ] Artboard's label should be 15px regardless of the scale of the Canvas
- [ ] Artboard cannot be originated when inside another Artboard
- [ ] Dragging and dropping an item outside of an Artboard should remove it from the Artboard's children
- [ ] Moving an item inside an artboard should make the item a child of the Artboard
- [ ] The layers panel should show the children of an artboard
- [ ] Hide & Lock should work also on Artboard layer

## This PR fixes/implements the following **bugs/features**:
- Implements (partially) #122
